### ISSUE TITLE
Changing `Chunk Output Inline` -> `Chunk Output in Console` breaks Rcpp chunks

### ISSUE THREAD

**Issue by DavisVaughan** [rstudio/rstudio#2501]
tl;dr) Try reproducing this: Open a new Rmd. Change `Chunk Output Inline` to `Chunk Output in Console`. Add a Rcpp chunk and try to execute the code below. It should fail.

```
#include <Rcpp.h>
using namespace Rcpp;

// [[Rcpp::export]]
NumericVector timesTwo(NumericVector x) {
  return x * 2;
}
```

---

I often enjoy using Chunk Output in Console because I find it a bit faster to work with. Below are before and after images of switching from `Chunk Output Inline` to `Chunk Output in Console` with Rcpp and Python chunks. I am using the daily build, 1.2.473.

<img width="1105" alt="screen shot 2018-03-25 at 5 40 24 pm" src="https://user-images.githubusercontent.com/19150088/37880318-4e94fb1c-3054-11e8-93ed-f7f139e89243.png">

<img width="1056" alt="screen shot 2018-03-25 at 5 40 40 pm" src="https://user-images.githubusercontent.com/19150088/37880320-51d6d99e-3054-11e8-9711-468db6116907.png">

You can see that the three symbols in the top right hand corner of the chunks disappeared. I think this is fine / expected for Python, and things work as expected. 

With python:
- With `Chunk Output Inline` we can run things line by line and use `print(df)` to get inline output (side note, shouldn't the line `df` alone output inline too? It doesn't currently).
- With `Chunk Output in Console` a python repl starts up in the console and code is sent down there when you run it line by line. This is great.

With Rcpp:
- With `Chunk Output Inline` you can source the whole file at once with the green arrow or Shift+Command+Enter.
- With `Chunk Output in Console` the green arrow is gone, and Shift+Command+Enter gives me errors such as `Error: unexpected symbol in "using namespace"`.

I think with Rcpp you should still be able to source chunks, right? Just send the output of the chunks (any errors) to the console alone, rather than have them in both the console and inline.

<details>

``````
---
title: "Untitled"
author: "Davis Vaughan"
date: "3/25/2018"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r}
library(reticulate)
```

```{Rcpp}
#include <Rcpp.h>
using namespace Rcpp;

// [[Rcpp::export]]
NumericVector timesTwo(NumericVector x) {
  return x * 2;
}
```

```{python}
import pandas as pd
import numpy as np
dates = pd.date_range('20130101', periods=6)
df = pd.DataFrame(np.random.randn(6,4), index=dates, columns=list('ABCD'))
df
```

```{r}
py$df
```


``````

</details>

**Comment by DavisVaughan** [comment]
Keep alive

**Comment by DavisVaughan** [comment]
Keep alive

---

An update on this is that, with RStudio 2022.02.0 Build 427, we do see the green right arrow when switching to `Chunk Output in Console` but now when I hit that green arrow it sends the chunk to the R console like it is R code:

With `Chunk Output Inline`:

<img width="1494" alt="Screen Shot 2022-03-04 at 10 11 36 AM" src="https://user-images.githubusercontent.com/19150088/156789019-4661fa5b-bbc8-4926-89f4-e087ee1d0f11.png">

With `Chunk Output in Console`:

<img width="1390" alt="Screen Shot 2022-03-04 at 10 13 07 AM" src="https://user-images.githubusercontent.com/19150088/156789080-1fcf260a-09e3-4782-9f15-054296135df1.png">

**Comment by jgutman** [comment]
Another issue in this area @mikebessuille -- at some point I think we need to prioritize some refactor of "notebook mode" aka Show chunk output inline vs in console in order to fix the collection of associated bugs here

**Comment by romainfrancois** [comment]
When "Show Output in console" is set (and the file is saved) I no longer see the play button: 

<img width="825" alt="image" src="https://user-images.githubusercontent.com/2625526/176437956-73c445f9-1c35-4b07-80c0-4680e92c2eb7.png">

Maybe instead it should be made to send a `sourceCpp` call, e.g. 

```r
Rcpp::sourceCpp(code = r"(#include <Rcpp.h>
using namespace Rcpp;

// [[Rcpp::export]]
NumericVector timesTwo(NumericVector x) {
  return x * 2;
})")
```

**Comment by jgutman** [comment]
That's a good idea- there's a related issue (I'll track down the issue number later) when chunk output in console is set, bash/python/sql etc chunks are sent to the console, where they fail, so this is closely related

**Comment by ddelzell** [comment]
+1 this Epic.  I interweave Python and R chunks constantly, and I NEVER like to have my output inline, it clutters up my code.  I would really like to be able to hit "run all" and have the console run repl_python() so that my python code chunks execute as python code.
